tapgarden/planter.go

```diff
@@ -2431,13 +2431,69 @@ func (c *ChainPlanter) sealBatch(ctx context.Context, params SealParams,
 		assetGroups = append(assetGroups, newGroup)
 	}
 
-	// Populate the group info for each seedling, to display to the caller.
+	// Populate the group info for each seedling.
 	batchWithGroupInfo := workingBatch.Copy()
 	for _, group := range assetGroups {
 		assetName := group.Genesis.Tag
 		batchWithGroupInfo.Seedlings[assetName].GroupInfo = group
 	}
 
+	// If universe commitments are enabled, the mint anchor pre-commitment
+	// output stored in the database must be updated with the associated
+	// newly generated group information.
+	if batchWithGroupInfo.UniverseCommitments {
+		if batchWithGroupInfo.GenesisPacket == nil {
+			return nil, fmt.Errorf("batch genesis packet is " +
+				"unexpectedly nil, cannot update mint anchor " +
+				"pre-commitment output")
+		}
+
+		// Fetch the pre-commit group key that was added to the
+		// modified batch.
+		groupKeyOpt, err := fetchPreCommitGroupKey(batchWithGroupInfo)
+		if err != nil {
+			return nil, fmt.Errorf("unable to fetch pre-commit "+
+				"group key: %w", err)
+		}
+
+		groupKey, err := groupKeyOpt.UnwrapOrErr(
+			fmt.Errorf("pre-commitment output group key is " +
+				"unexpectedly absent"),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		// Ensure that the group key is set in the genesis packet
+		// pre-commitment output descriptor.
+		fundedAnchor := batchWithGroupInfo.GenesisPacket
+		if fundedAnchor == nil {
+			return nil, fmt.Errorf("funded anchor is " +
+				"unexpectedly nil, cannot update mint anchor " +
+				"pre-commitment output descriptor")
+		}
+
+		preCommitDesc, err :=
+			fundedAnchor.PreCommitmentOutput.UnwrapOrErr(
+				fmt.Errorf("pre-commitment output is " +
+					"unexpectedly absent"),
+			)
+		if err != nil {
+			return nil, err
+		}
+
+		if preCommitDesc.GroupPubKey.IsSome() {
+			return nil, fmt.Errorf("pre-commitment output " +
+				"group key is unexpectedly set, cannot " +
+				"update mint anchor pre-commitment output")
+		}
+
+		preCommitDesc.GroupPubKey = fn.Some(groupKey)
```

Comment:
A this point, won't the group key _always_ be known? Since this is the OG group key and not one of the tweaked versions that we use to ensure commitment uniqueness elsewhere.